### DISCUSSION TITLE
Please help me getting a custom action to work

### DISCUSSION THREAD

**Discussion by David-Else** [focustimerhq/FocusTimer#681] [Q&A]
Hi, I have tried everything to get the network to turn off when i am doing a pomodoro. I don't see what I am doing wrong. I have also tried plain `nmcli networking off` as the command. The command works fine when typed into the terminal. I tried some example commands I found in the issues here and they did nothing either.

![image](https://github.com/gnome-pomodoro/gnome-pomodoro/assets/12832280/8215d670-1318-472f-8982-3c4a0c9a0761)

I am using the built-in Debian 12 apt version 0.22.1

**Comment by kamilprusko** [comment]
You need to define two actions:

![Screenshot from 2023-11-22 13-28-56](https://github.com/gnome-pomodoro/gnome-pomodoro/assets/1135423/1d9b1027-db92-4cfc-afdf-f8133e20509c)
![Screenshot from 2023-11-22 13-29-04](https://github.com/gnome-pomodoro/gnome-pomodoro/assets/1135423/9ddbe769-10f2-406a-9ada-3bb038098763)

I would be weary disabling whole network manager as it may interfere with regular work (docker, virtual machines, local web development)... I'm not an expert in that topic :)

**Comment by David-Else** [comment]
Brilliant, thanks, that works!

> I would be weary disabling whole network manager as it may interfere with regular work (docker, virtual machines, local web development)... 

You are right. I only really want to disable browsing the internet so I came up with these two scripts to call:
`block-web-surfing`
```
#!/bin/bash
set -e

# Block web traffic ports
sudo iptables -A OUTPUT -p tcp --dport 80 -j REJECT
sudo iptables -A OUTPUT -p tcp --dport 443 -j REJECT
notify-send "Web surfing blocked on ports 80 and 443"
```
and `unblock-web-surfing`
```
#!/bin/bash
set -e

sudo iptables -D OUTPUT -p tcp --dport 80 -j REJECT
sudo iptables -D OUTPUT -p tcp --dport 443 -j REJECT
notify-send "Web surfing unblocked on ports 80 and 443"
```
But that didn't work in Pomodoro, I assume because the `sudo` is not allowed?

I tried these as direct commands inside your app but no luck:
```
sudo iptables -A OUTPUT -p tcp --dport 80 -j REJECT && sudo iptables -A OUTPUT -p tcp --dport 443 -j REJECT && notify-send "Web surfing blocked on ports 80 and 443"

sudo iptables -D OUTPUT -p tcp --dport 80 -j REJECT && sudo iptables -D OUTPUT -p tcp --dport 443 -j REJECT && notify-send "Web surfing unblocked on ports 80 and 443"
```
Any ideas? Cheers!
